src/components/Observations/zone/ReplaceObservationPlotModal.tsx

```diff
@@ -0,0 +1,148 @@
+import { useEffect, useState, useMemo } from 'react';
+import { Grid, Typography, useTheme } from '@mui/material';
+import { Button, BusySpinner, DialogBox, Dropdown, Textfield } from '@terraware/web-components';
+import strings from 'src/strings';
+import useSnackbar from 'src/utils/useSnackbar';
+import { ObservationMonitoringPlotResultsPayload, ReplaceObservationPlotDuration } from 'src/types/Observations';
+import { useLocalization, useOrganization } from 'src/providers';
+import { useAppSelector, useAppDispatch } from 'src/redux/store';
+import { selectReplaceObservationPlot } from 'src/redux/features/observations/observationsSelectors';
+import { requestReplaceObservationPlot } from 'src/redux/features/observations/observationsAsyncThunks';
+import { requestObservationsResults } from 'src/redux/features/observations/observationsThunks';
+
+export interface ReplaceObservationPlotModalProps {
+  onClose: () => void;
+  observationId: number;
+  monitoringPlot: ObservationMonitoringPlotResultsPayload;
+}
+
+export default function ReplaceObservationPlotModal(props: ReplaceObservationPlotModalProps): JSX.Element {
+  const { onClose, observationId, monitoringPlot } = props;
+  const dispatch = useAppDispatch();
+  const { activeLocale } = useLocalization();
+  const { selectedOrganization } = useOrganization();
+  const snackbar = useSnackbar();
+  const theme = useTheme();
+  const [justification, setJustification] = useState<string>();
+  const [duration, setDuration] = useState<ReplaceObservationPlotDuration | undefined>();
+  const [validate, setValidate] = useState<boolean>(false);
+  const [requestId, setRequestId] = useState<string>('');
+
+  const result = useAppSelector((state) => selectReplaceObservationPlot(state, requestId));
+
+  const replaceObservationPlot = () => {
```

Comment:
Does this need to call onClose?

Reply:
Thanks for the review Charlie. `replaceObservationPlot` function dispatches the request.
We have a listener on the result selector, which calls `onClose` if the request succeeded.
This modal handles the entire request/response handling.
Will be reused in the list view and the map view (next effort).